Almost Working Autoscaling and Spot Instance Usage

salvis2 opened the pull request:
Spot Instance Usage is working for the user-spot and worker-spot groups. They are deployed at 0 nodes by default and *should* have all the right tags / labels / taints for autoscaling. Should resolve 
#6 and #7 .

Autoscaling up is not quite working. I can't exactly follow the example [here](https://github.com/terraform-aws-modules/terraform-aws-eks/tree/master/examples/irsa) because this example wants to deploy the cluster-autoscaler via helm. I want to deploy it with the Terraform helm provider, but I think I am missing something. 

Logging into the JupyterHub triggers autoscaling (according to the logs), but the node does not spin up within 10 minutes and the user pod spawning fails with a timeout error. Begins addressing 

That being said, the JupyterHub seems to be set up fine (as far I can tell without logging in).

salvis2 committed:
Added in worker group templates for autoscaling and spot instancing

salvis2 committed:
Change vpc name to be a variable

salvis2 committed:
Formatting

salvis2 committed:
Delete unnecessary newline characters in multi-line strings

salvis2 committed:
Formatting

salvis2 committed:
Pin to specific kubernetes provider version. Breaking change in later versions.

salvis2 committed:
Correct syntax for terraform apply

salvis2 committed:
Added = to update-kubeconfig call

salvis2 committed:
Additional tags to enable autoscaling

salvis2 committed:
Initial work on enabling the cluster-autoscaler.

salvis2 committed:
Corrected autoscaling tags

salvis2 committed:
Propogate tags at launch

salvis2 committed:
Have to explicitly declare ASG min size as 0

salvis2 committed:
Formatting

salvis2 committed:
Delete duplicate resources

salvis2 committed:
Corrected autoscaling options

salvis2 committed:
Changes for my config

salvis2 committed:
Made VPC name variable

salvis2 committed:
Added variables for cluster-autoscaler

salvis2 committed:
VPC variable name for me

salvis2 committed:
Set core node label

salvis2 committed:
Don't need tag propagation when default size is 0

salvis2 committed:
Formatting

salvis2 committed:
Default JupyterHub values that I've used

yuvipanda commented on aws/variables.tf:
```diff
@@ -1,14 +1,18 @@
 
 variable "cluster_name" {
-  default = "test-eks-cluster-1"
+  default = "eks-cluster-salvis"
```
We should leave more generic defaults here, and put overrides in a tfvars file.

yuvipanda commented on aws/values.yaml:
```diff
@@ -1,17 +1,55 @@
+      storageClass: aws-efs
+
+hub:
+  resources: # should fill up single node to near capacity
```
I think the hub needs far less resources, we should definitely not take up a full single node with it.

This is what I see on one of our largest hubs:

```
    resources:
      limits:
        memory: 1Gi
      requests:
        cpu: 200m
        memory: 256Mi
```

yuvipanda committed:
Request fewer resources for hub pod

yuvipanda committed:
cluster-autoscaler namespace & service acct shouldn't be configurable

We should limit configuration options to things we want users
to change.

yuvipanda committed:
Reset variable defaults

yuvipanda committed:
Consolidate cluster autoscaler config

yuvipanda committed:
Change name of template tfvars

.tf files seem to be all read by terraform, causing confusion
and big error messages
